solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,400 @@
+# VPC Variables
+##############################################################################
+
+variable "existing_vpc_id" {
+  description = "The ID of an existing VPC."
+  type        = string
+}
+
+variable "existing_subnet_id" {
+  description = "The ID of an existing subnet"
+  type        = string
+}
+
+##############################################################################
+
+
+##############################################################################
+# VSI Variables
+##############################################################################
+
+variable "vsi_name" {
+  description = "The name of the Virtual server instance."
+  type        = string
+  default     = "vsi"
+}
+
+variable "image_id" {
+  description = "Image ID used for VSI. Run 'ibmcloud is images' to find available images in a region."
+  type        = string
+  default     = "r006-cc341965-a523-464e-969f-391e2661c125"
+}
+
+variable "ssh_public_key" {
+  description = "A public SSH Key for VSI creation which does not already exist in the deployment region. Must be an RSA key with a key size of either 2048 bits or 4096 bits (recommended) - See https://cloud.ibm.com/docs/vpc?topic=vpc-ssh-keys. To use an existing key, enter a value for the variable 'existing_ssh_key_name' instead."
+  type        = string
+  default     = null
+  validation {
+    error_message = "Public SSH Key must be a valid ssh rsa public key."
+    condition     = var.ssh_public_key == null || can(regex("ssh-rsa AAAA[0-9A-Za-z+/]+[=]{0,3} ?([^@]+@[^@]+)?", var.ssh_public_key))
+  }
+  validation {
+    condition     = var.ssh_public_key != null || length(var.existing_ssh_key_ids) > 0 ? true : false
+    error_message = "Please provide a value for either `ssh_public_key` or `existing_ssh_key_ids`."
+  }
+}
+
+variable "existing_ssh_key_ids" {
+  description = "IDs of existing SSH keys to use while creating VSI."
+  type        = list(string)
+  default     = []
+}
+
+variable "machine_type" {
+  description = "VSI machine type. Run 'ibmcloud is instance-profiles' to get a list of regional profiles"
+  type        = string
+  default     = "cx2-2x4"
+}
+
+variable "user_data" {
+  description = "User data to initialize VSI deployment"
+  type        = string
+  default     = null
+}
+
+variable "use_boot_volume_key_as_default" {
+  description = "Set to true to use the key specified in the `boot_volume_encryption_key` input as default for all volumes, overriding any key value that may be specified in the `encryption_key` option of the `block_storage_volumes` input variable. If set to `false`,  the value passed for the `encryption_key` option of the `block_storage_volumes` will be used instead."
+  type        = bool
+  default     = false
+}
+
+variable "boot_volume_encryption_key" {
+  description = "CRN of boot volume encryption key"
+  default     = null
+  type        = string
+}
+
+variable "existing_kms_instance_guid" {
```

Comment:
We should ask for CRN to be consistent with other DAs. And it would only be needed if user was not supplying existing KMS key so suggest to add validation for that.

Also suggest to validate the CRN value (see [example](https://github.com/terraform-ibm-modules/terraform-ibm-scc/blob/8ca8ad09a6e1289d6a46594c6ddae86e86f4d0a8/solutions/fully-configurable/variables.tf#L196-L202))